pgmpy/inference/mplp.py

```diff
@@ -0,0 +1,752 @@
+        >>> factor_a   = Factor(['A'], cardinality=[2], value=np.array([0.54577, 1.8323]))
+        >>> factor_b   = Factor(['B'], cardinality=[2], value=np.array([0.93894, 1.065]))
+        >>> factor_c   = Factor(['C'], cardinality=[2], value=np.array([0.89205, 1.121]))
+        >>> factor_d   = Factor(['D'], cardinality=[2], value=np.array([0.56292, 1.7765]))
+        >>> factor_e   = Factor(['E'], cardinality=[2], value=np.array([0.47117, 2.1224]))
+        >>> factor_f   = Factor(['F'], cardinality=[2], value=np.array([1.5093, 0.66257]))
+        >>> factor_g   = Factor(['G'], cardinality=[2], value=np.array([0.48011, 2.0828]))
+        >>> factor_h   = Factor(['H'], cardinality=[2], value=np.array([2.6332, 0.37977]))
+        >>> factor_i   = Factor(['I'], cardinality=[2], value=np.array([1.992, 0.50202]))
+        >>> factor_j   = Factor(['J'], cardinality=[2], value=np.array([1.6443, 0.60817]))
+        >>> factor_k   = Factor(['K'], cardinality=[2], value=np.array([0.39909, 2.5057]))
+        >>> factor_l   = Factor(['L'], cardinality=[2], value=np.array([1.9965, 0.50087]))
+        >>> factor_m   = Factor(['M'], cardinality=[2], value=np.array([2.4581, 0.40681]))
+        >>> factor_n   = Factor(['N'], cardinality=[2], value=np.array([2.0481, 0.48826]))
+        >>> factor_o   = Factor(['O'], cardinality=[2], value=np.array([0.6477, 1.5439]))
+        >>> factor_p   = Factor(['P'], cardinality=[2], value=np.array([0.93844, 1.0656]))
+        >>> factor_a_b = Factor(['A', 'B'], cardinality=[2, 2], value=np.array([1.3207, 0.75717, 0.75717, 1.3207]))
+        >>> factor_b_c = Factor(['B', 'C'], cardinality=[2, 2], value=np.array([0.00024189, 4134.2, 4134.2, 0.000241]))
+        >>> factor_c_d = Factor(['C', 'D'], cardinality=[2, 2], value=np.array([0.0043227, 231.34, 231.34, 0.0043227]))
+        >>> factor_e_f = Factor(['E', 'F'], cardinality=[2, 2], value=np.array([31.228, 0.032023, 0.032023, 31.228]))
+        >>> student.add_factors(factor_a, factor_b, factor_c, factor_d,
+        ...            factor_e, factor_f, factor_g, factor_h,
+        ...            factor_i, factor_j, factor_k, factor_l,
+        ...            factor_m, factor_n, factor_o, factor_p,
+        ...            factor_a_b, factor_b_c, factor_c_d, factor_d_e,
+        >>> mplp = MPLP(student)
+        >>> mplp.map_query(1000, 100, 20, 0.0002, 0.0002)
+        """
+
+        self.run_mplp(max_iterations, objective_threshold, int_gap_threshold)
+        for i in range(max_tightening):
+            int_gap = self.last_obj - self.m_best_val
+            if int_gap < int_gap_threshold:
+                break
+            if int_gap < 1:
+                n_iter_later = 600
+            ncluster_added = self.tighten_cycle()
+            if ncluster_added == 0:
+                print("This is the best result")
+                break
+            self.run_mplp(n_iter_later, objective_threshold, int_gap_threshold)
+
+
+class Region:
```

Comment:
I don't think ```Region``` class belongs here. Please update the documentation so that we can decide where to move this class.